specification/cognitiveservices/data-plane/Speech/SpeechToText/preview/v3.1-preview.1/speechtotext.json

```diff
@@ -0,0 +1,6409 @@
+      "type": "object",
+      "properties": {
+        "code": {
+          "$ref": "#/definitions/DetailedErrorCode"
+        },
+        "details": {
+          "description": "Additional supportive details regarding the error and/or expected policies.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "message": {
+          "description": "High level error message.",
+          "type": "string"
+        },
+        "target": {
+          "description": "The source of the error.\r\nFor example it would be \"documents\" or \"document id\" in case of invalid document.",
+          "type": "string"
+        },
+        "innerError": {
+          "$ref": "#/definitions/InnerError"
+        }
+      }
+    },
+    "LanguageIdentificationProperties": {
+      "title": "LanguageIdentificationProperties",
+      "required": [
+        "candidateLocales"
+      ],
+      "type": "object",
+      "properties": {
+        "candidateLocales": {
+          "description": "The candidate locales for language identification (example [\"en-US\", \"de-DE\", \"es-ES\"]). A minimum of 2 and a maximum of 10 candidate locales, including the main locale for the transcription, is supported.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "speechModelMapping": {
+          "description": "An optional mapping of locales to speech model entities. If no model is given for a locale, the default base model is used.\r\nKeys must be locales contained in the candidate locales, values are entities for models of the respective locales.",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/EntityReference"
+          }
+        }
+      }
+    },
+    "ModelAdaptableWithDatasetKinds": {
+      "title": "ModelAdaptableWithDatasetKinds",
+      "type": "object",
+      "properties": {
+        "language": {
+          "description": "A value indicating whether adaptation with text (language dataset) is supported.",
+          "type": "boolean",
+          "readOnly": true
+        },
+        "acoustic": {
+          "description": "A value indicating whether adaptation with audio and transcription (acoustic dataset) is supported.",
+          "type": "boolean",
+          "readOnly": true
+        },
+        "pronunciation": {
+          "description": "A value indicating whether adaptation with pronunciation dataset is supported.",
+          "type": "boolean",
+          "readOnly": true
+        },
+        "audioFiles": {
```

Comment:
As a boolean, this isn't very intuitively named. For booleans, could you include something like "has" as a prefix e.g., `hasAudioFiles`. Same with other booleans here as well. This would likely be a red flag in any SDK API review.

Reply:
In the meeting, we discussed this and recommended something like `audioFilesSupported`, etc. Array of string values might also work if consistent values with type names is necessary, but the dev experience won't be as nice as setting booleans.